kong/observability/tracing/propagation/extractors/instana.lua

```diff
@@ -0,0 +1,67 @@
+local _EXTRACTOR        = require "kong.observability.tracing.propagation.extractors._base"
+local propagation_utils = require "kong.observability.tracing.propagation.utils"
+local from_hex          = propagation_utils.from_hex
+
+local INSTANA_EXTRACTOR = _EXTRACTOR:new({
+  headers_validate = {
+    any = {
+      "x-instana-t",
+      "x-instana-s",
+      "x-instana-l",
+    }
+  }
+})
+
+function INSTANA_EXTRACTOR:get_context(headers)
+
+  local trace_id_raw = headers["x-instana-t"]
+
+  if type(trace_id_raw) ~= "string" then
+    return
+  end
+
+  if trace_id_raw then
+    trace_id_raw = trace_id_raw:match("^(%x+)")
+    local trace_id_len = trace_id_raw and #trace_id_raw or 0
+    if not trace_id_raw or
+       not(trace_id_len == 16 or trace_id_len == 32)
+    then
+      kong.log.warn("x-instana-t header invalid; ignoring.")
+    end
+  end
+
+  local span_id_raw = headers["x-instana-s"]
+
+  if type(span_id_raw) ~= "string" then
+    return
+  end
+
+  if span_id_raw then
+    span_id_raw = span_id_raw:match("^(%x+)")
+    if not span_id_raw then
+      kong.log.warn("x-instana-s header invalid; ignoring.")
+    end
+  end
+
+  local level_id_raw = headers["x-instana-l"]
+
+  if level_id_raw then
```

Comment:
same

Reply:
We probably still need this one since level id can be nil.

Reply:
oh yeah, sorry, ignore this comment